Scala Example Project Build Fails: sbt "runMain standaloneExample" fails to build

### Describe the bug

System: MacOS Intel

There's actually two issues, but I resolved one rather easily. First, there are no intel jars published to maven for macos. To resolve that, I build the client shadowJar in the java project:

```
./gradlew clean && ./gradlew :client:shadowJar
```
This produces a client jar at ./client/build/libs/client-osx-x86_64.jar

I reference that jar in the examples/scala/build.sbt instead of the maven coordinates:

```
unmanagedJars in Compile += file("/Users/jkyle/Projects/valkey-glide/java/client/build/libs/client-osx-x86_64.jar")
```

Then build the project with

```
sbt "runMain StandaloneExample"
```

This produces the following build errors:

```
❯ sbt "runMain StandaloneExample"
[info] welcome to sbt 1.10.1 (Eclipse Adoptium Java 21.0.6)
[info] loading project definition from /Users/jkyle/Projects/valkey-glide/examples/scala/project
[info] loading settings for project root from build.sbt ...
[info] set current project to example (in build file:/Users/jkyle/Projects/valkey-glide/examples/scala/)
[info] compiling 2 Scala sources to /Users/jkyle/Projects/valkey-glide/examples/scala/target/scala-3.3.3/classes ...
[error] -- [E008] Not Found Error: /Users/jkyle/Projects/valkey-glide/examples/scala/src/main/scala/ClusterExample.scala:64:28
[error] 64 |                InfoOptions.builder()
[error]    |                ^^^^^^^^^^^^^^^^^^^
[error]    |value builder is not a member of object glide.api.models.commands.InfoOptions
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 3 s, completed Mar 20, 2025, 10:28:53 AM
```

I took a glance at the InfoOptions class in the java project and the builder, `./client/src/main/java/glide/api/models/commands/InfoOptions.java`, and it looks like a thin wrapper around the Sections enum.  The builder functionality seems to have moved into the GlideClient.

I was able to resolve this by fixing the call to InfoOptions.Sections in the cluster client. This produced a protobuf version error in the java client jar I built. Fix that too by bumping the client.

Which then led to 

```
[error] java.lang.RuntimeException: Cannot load Netty native components for the current os version and arch: Mac OS X 15.3.2 x86_64.
[error]
[error]         at glide.connectors.resources.Platform.getThreadPoolResourceSupplier(Platform.java:100)
[error]         at glide.api.BaseClient.createClient(BaseClient.java:364)
[error]         at glide.api.GlideClient.createClient(GlideClient.java:90)
[error]         at StandaloneExample$.createClient(StandaloneExample.scala:41)
[error]         at StandaloneExample$.StandaloneExample$$$_$loop$1(StandaloneExample.scala:70)
[error]         at StandaloneExample$.execAppLogic(StandaloneExample.scala:108)
[error]         at StandaloneExample$.main(StandaloneExample.scala:125)
[error]         at StandaloneExample.main(StandaloneExample.scala)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] stack trace is suppressed; run last Compile / runMain for the full output
[error] (Compile / runMain) Cannot load Netty native components for the current os version and arch: Mac OS X 15.3.2 x86_64.
[error] Total time: 5 s, completed Mar 20, 2025, 11:16:39 AM
```

I fiddled around with the netty dependencies trying to get those working for the arch. . . . 
but raises the question of whether Intel macs are completely unsupported?

### Expected Behavior

Expected behavior is a clean build and test client execution in `examples/scala` when on Intel macs.

### Current Behavior

Currently throws build errors due to API mismatch between java API and scala example and missing library dependencies for x86_64 arch targets.

### Reproduction Steps

```
cd examples/scala && sbt "runMain StandaloneExample"
```

### Possible Solution

Looks like a combination of 

* updating the scala cluster client example code for the new InfoOptions class api
* updating library dependencies in the java client for x86_64 macos targets
* building and publishing library jars for valkey-client for x86_64 macos arch

### Additional Information/Context

_No response_

### Client version used

HEAD, sha: 581bae10

### Engine type and version

valkey

### OS

MacOS, x86_64

### Language

Java

### Language Version

21

### Cluster information

Didn't get that far.

### Logs

Initial failure produces these logs when building scala client

```
[error] lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[error] https://repo1.maven.org/maven2/io/valkey/valkey-glide/1.3.1/valkey-glide-1.3.1-osx-x86_64.jar: not found: https://repo1.maven.org/maven2/io/valkey/valkey-glide/1.3.1/valkey-glide-1.3.1-osx-x86_64.jar
[error]
[error]         at lmcoursier.internal.shaded.coursier.Artifacts$.$anonfun$fetchArtifacts$9(Artifacts.scala:365)
[error]         at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$extension$1(Task.scala:14)
[error]         at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$extension$1$adapted(Task.scala:14)
[error]         at lmcoursier.internal.shaded.coursier.util.Task$.wrap(Task.scala:82)
[error]         at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$2(Task.scala:14)
[error]         at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
[error]         at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:51)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:74)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error]         at java.base/java.lang.Thread.run(Thread.java:1583)
[error] Caused by: lmcoursier.internal.shaded.coursier.cache.ArtifactError$NotFound: not found: https://repo1.maven.org/maven2/io/valkey/valkey-glide/1.3.1/valkey-glide-1.3.1-osx-x86_64.jar
[error]         at lmcoursier.internal.shaded.coursier.cache.internal.Downloader.checkErrFile$1(Downloader.scala:471)
[error]         at lmcoursier.internal.shaded.coursier.cache.internal.Downloader.$anonfun$shouldDownload$16(Downloader.scala:518)
[error]         at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[error]         at scala.util.Success.$anonfun$map$1(Try.scala:255)
[error]         at scala.util.Success.map(Try.scala:213)
[error]         at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:42)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:74)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error]         at java.base/java.lang.Thread.run(Thread.java:1583)
[error] (update) lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[error] https://repo1.maven.org/maven2/io/valkey/valkey-glide/1.3.1/valkey-glide-1.3.1-osx-x86_64.jar: not found: https://repo1.maven.org/maven2/io/valkey/valkey-glide/1.3.1/valkey-glide-1.3.1-osx-x86_64.jar
```



### Other information

I did checkout `v1.3.1` tag and test there to validate it's not something just on ^HEAD.

--- a/examples/scala/build.sbt
+++ b/examples/scala/build.sbt
@@ -12,9 +12,9 @@ val os = System.getProperty("os.name").toLowerCase
 val platformClassifier = {
   (os, System.getProperty("os.arch").toLowerCase) match {
     case (mac, arm) if mac.contains("mac") && (arm.contains("aarch") || arm.contains("arm")) => "osx-aarch_64"
-    case (mac, x86) if mac.contains("mac") && x86.contains("x86") => "osx-x86_64"
+    case (mac, x86) if mac.contains("mac") && (x86.contains("x86") || x86.contains("amd64")) => "osx-x86_64"
     case (linux, arm) if linux.contains("linux") && (arm.contains("aarch") || arm.contains("arm")) => "linux-aarch_64"
-    case (linux, x86) if linux.contains("linux") && x86.contains("x86") => "linux-x86_64"
+    case (linux, x86) if linux.contains("linux") && (x86.contains("x86") || x86.contains("amd64")) => "linux-x86_64"
     case (osName, archName) => throw new RuntimeException(s"Unsupported platform $osName $archName")
   }
 }

--- a/examples/scala/src/main/scala/ClusterExample.scala
+++ b/examples/scala/src/main/scala/ClusterExample.scala
@@ -26,7 +26,7 @@ object ClusterExample {
      *
      * @return An instance of <code>Future[GlideClusterClient]</code> connected to the specified nodes.
      */
-    private def createClient(nodesList: List[(String, Int)] = List(("localhost", 6379))): Future[GlideClusterClient] = {
+    private def createClient(nodesList: List[(String, Int)] = List(("localhost", 7000))): Future[GlideClusterClient] = {
         // Check `GlideClientConfiguration` for additional options.
         val config = GlideClusterClientConfiguration.builder()
             .addresses(nodesList.map((host, port) => NodeAddress.builder().host(host).port(port).build()).asJava)
@@ -67,7 +67,7 @@ object ClusterExample {
             _ = Logger.log(
                 Logger.Level.INFO,
                 "app",
-                "INFO REPLICATION responses from all nodes are=\n$infoReplResps",
+                s"INFO REPLICATION responses from all nodes are\n${infoReplResps.getMultiValue()}",
             )
         } yield ()
     }

--- a/examples/scala/src/main/scala/ClusterExample.scala
+++ b/examples/scala/src/main/scala/ClusterExample.scala
@@ -26,7 +26,7 @@ object ClusterExample {
      *
      * @return An instance of <code>Future[GlideClusterClient]</code> connected to the specified nodes.
      */
-    private def createClient(nodesList: List[(String, Int)] = List(("localhost", 7000))): Future[GlideClusterClient] = {
+    private def createClient(nodesList: List[(String, Int)] = List(("localhost", 6379))): Future[GlideClusterClient] = {
         // Check `GlideClientConfiguration` for additional options.
         val config = GlideClusterClientConfiguration.builder()
             .addresses(nodesList.map((host, port) => NodeAddress.builder().host(host).port(port).build()).asJava)
@@ -67,7 +67,7 @@ object ClusterExample {
             _ = Logger.log(
                 Logger.Level.INFO,
                 "app",
-                s"INFO REPLICATION responses from all nodes are\n${infoReplResps.getMultiValue()}",
+                s"INFO REPLICATION responses from all nodes are:\n${infoReplResps.getMultiValue()}",
             )
         } yield ()
     }

--- a/examples/scala/src/main/scala/ClusterExample.scala
+++ b/examples/scala/src/main/scala/ClusterExample.scala
@@ -1,6 +1,6 @@
 import glide.api.GlideClusterClient
 import glide.api.logging.Logger
-import glide.api.models.commands.InfoOptions
+import glide.api.models.commands.InfoOptions.Section
 import glide.api.models.configuration.{GlideClusterClientConfiguration, NodeAddress}
 import glide.api.models.configuration.RequestRoutingConfiguration.SimpleMultiNodeRoute.ALL_NODES
 import glide.api.models.exceptions.{ClosingException, ConnectionException, ExecAbortException, TimeoutException}
@@ -61,9 +61,7 @@ object ClusterExample {
 
             // Send INFO REPLICATION with routing option to all nodes
             infoReplResps <- client.info(
-                InfoOptions.builder()
-                    .section(InfoOptions.Section.REPLICATION)
-                    .build(),
+                Array(Section.REPLICATION),
                 ALL_NODES
             ).asScala
             _ = Logger.log(

